pkg/kubelet/cm/cpumanager/policy_options.go

```diff
@@ -56,6 +58,14 @@ func CheckPolicyOptionAvailable(option string) error {
 		return fmt.Errorf("unknown CPU Manager Policy option: %q", option)
 	}
 
+	// must override the base feature gate check. Relevant only for alpha (disabled by default).
+	// for beta options are enabled by default and we totally want to keep the possibility to
+	// disable them explicitly.
+	if alphaOptions.Has(option) && checkPolicyOptionHasEnablementFile(option) {
```

Comment:
wouldn't this put the cluster in TPNU?

Reply:
this is the internal check: `alphaOptions` is the internal set of known cpumanager policy options, and this specific guard is meant to bypass the upstream alpha FG check for features which have the enablement file. It was implemented this way (vs extending the existing guard below) to make it as easy as possible to remove later on when the carry is no longer needed. It is meant to be equivalent to:
```
// CheckPolicyOptionAvailable verifies if the given option can be used depending on the Feature Gate Settings.
// returns nil on success, or an error describing the failure on error.
func CheckPolicyOptionAvailable(option string) error {
	if !alphaOptions.Has(option) && !betaOptions.Has(option) && !stableOptions.Has(option) {
		return fmt.Errorf("unknown CPU Manager Policy option: %q", option)
	}

	if alphaOptions.Has(option) {
		if checkPolicyOptionHasEnablementFile(option) {
			return nil
		}
		if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CPUManagerPolicyAlphaOptions) {
			return fmt.Errorf("CPU Manager Policy Alpha-level Options not enabled, but option %q provided", option)
		}
	}

	if betaOptions.Has(option) && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CPUManagerPolicyBetaOptions) {
		return fmt.Errorf("CPU Manager Policy Beta-level Options not enabled, but option %q provided", option)
	}

	return nil
}
```